### DISCUSSION TITLE
Declare queues at container run

### DISCUSSION THREAD

**Discussion by paulomorgado** [rabbitmq/rabbitmq-server#10878] [Questions]
When running the `rabbitmq` container, what is the best practice to declare queues on container run?

**Comment by lukebakken** [comment]
Hi @paulomorgado! Your question would be best asked here: https://github.com/docker-library/rabbitmq/discussions

**Reply by mkuratczyk** [reply]
Some options:
https://www.rabbitmq.com/docs/definitions#import-on-boot
https://github.com/rabbitmq/messaging-topology-operator

**Reply by paulomorgado** [reply]
@mkuratczyk, how do I set this through environment variables set for the container?

```
# Does not require management plugin to be enabled.
definitions.import_backend = local_filesystem
definitions.local.path = /path/to/definitions/defs.json
```

**Reply by lukebakken** [reply]
You can't use environment variables for this. Use the `rabbitmq.conf` file, either via a mount, or via your own Docker image that uses the `rabbitmq` image as a base, that copies `rabbitmq.conf` and the definitions file into it.
